knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java

```diff
@@ -107,7 +108,7 @@ public void doService() {
 	// checks for path traversal attacks
 	private void checkRequiredFile(String fileName) {
 		File targetDirectory = GeneralUtilities.getPreviewFilesStorageDirectoryPath();
-		FileUtils.checkPathTraversalAttack(fileName, targetDirectory);
+		PathTraversalChecker.get(fileName, targetDirectory.getName());
```

Comment:
it doesn't make sense: inputs are reversed, since targetDirectory is the safe dir and filename is what has to be checked

Reply:
Fixed